commit bcae70d0f75b249ee1141ba25dc75cc9adbdf18b
Author: Dave Stevenson <dave.stevenson@raspberrypi.org>
Date:   Thu Jan 24 16:20:38 2019 +0000

    staging: bcm2835-camera: Correct ctrl min/max/step/def to 64bit
    
    The V4L2 control API was expanded to take 64 bit values in commit
    0ba2aeb6dab (Apr 16 2014), but as this driver wasn't in the mainline
    kernel at that point this was overlooked.
    
    Update to use 64 bit values. This also fixes a couple of warnings
    in 64 bit builds.
    
    Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.org>

diff --git a/drivers/staging/vc04_services/bcm2835-camera/controls.c b/drivers/staging/vc04_services/bcm2835-camera/controls.c
index e7c3c68e40f6..465e6d20b6b5 100644
--- a/drivers/staging/vc04_services/bcm2835-camera/controls.c
+++ b/drivers/staging/vc04_services/bcm2835-camera/controls.c
@@ -78,10 +78,10 @@ struct bm2835_mmal_v4l2_ctrl {
 	/* control minimum value or
 	 * mask for MMAL_CONTROL_TYPE_STD_MENU
 	 */
-	s32 min;
-	s32 max; /* maximum value of control */
-	s32 def;  /* default value of control */
-	s32 step; /* step size of the control */
+	s64 min;
+	s64 max; /* maximum value of control */
+	s64 def;  /* default value of control */
+	u64 step; /* step size of the control */
 	const s64 *imenu; /* integer menu array */
 	u32 mmal_id; /* mmal parameter id */
 	bm2835_mmal_v4l2_ctrl_cb *setter;
@@ -1244,7 +1244,7 @@ int bm2835_mmal_init_controls(struct bm2835_mmal_dev *dev,
 
 		case MMAL_CONTROL_TYPE_STD_MENU:
 		{
-			int mask = ctrl->min;
+			u64 mask = ctrl->min;
 
 			if (ctrl->id == V4L2_CID_SCENE_MODE) {
 				/* Special handling to work out the mask
@@ -1254,11 +1254,11 @@ int bm2835_mmal_init_controls(struct bm2835_mmal_dev *dev,
 				 */
 				int i;
 
-				mask = 1 << V4L2_SCENE_MODE_NONE;
+				mask = BIT(V4L2_SCENE_MODE_NONE);
 				for (i = 0;
 				     i < ARRAY_SIZE(scene_configs);
 				     i++) {
-					mask |= 1 << scene_configs[i].v4l2_scene;
+					mask |= BIT(scene_configs[i].v4l2_scene);
 				}
 				mask = ~mask;
 			}
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1331_bcae70d0f75b249ee1141ba25dc75cc9adbdf18b.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1331_bcae70d0f75b249ee1141ba25dc75cc9adbdf18b.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
